### ISSUE TITLE
Make `serve_book()` to support other html-based formats like pagedown::book_crc

### ISSUE THREAD

**Issue by jooyoungseo** [rstudio/bookdown#1051]
Currently, `bookdown::serve_book()` seems to ignore output format specified in `_output.yml` and to force live preview in `bookdown::gitbook()` format.

Could we make it work like `bookdown::render_book()` so to support other html-based format (e.g., `pagedown::book_crc`; `xaringan::moon_reader`)?

As you know, `xaringan::inf_mr()` is intended for a single Rmd file; we need the general live preview for html-based multiple Rmd files.

**Comment by cderv** [comment]
Yes currently `bookdown::serve_book()` only allow to preview the first HTML book format found in `_output.yaml` among a list of support own **bookdown** format
https://github.com/rstudio/bookdown/blob/f7d062298ef1a209e18cf422f2327780d647f3d3/R/utils.R#L374
https://github.com/rstudio/bookdown/blob/f7d062298ef1a209e18cf422f2327780d647f3d3/R/utils.R#L399-L406

So you would need to put the HTML to preview first in `_output..yml`, but it won't currently work with `pagegedown::book_crc()`

However, `bookdown::serve_book()` is specific to **bookdown** related format for book-like output. `pagedown::book_crc()` is not a multiple document format: it is a single document format, and is to be rendered with `rmarkdown::render()` and not `bookdown::render_book()`

So I would not use `pagedown::book_crc()` or `xaringan::moon_reader()` with `bookdown::render_book()`. 
Is it still a use case where you want to use a format aimed at a single document but write several Rmd sources and use `bookdown::render_book()` for its merging capabilities before rendering ? 
I recall you mentioned that in another issue.

Note about the feature request for `serve_book()`: this is related to #953 with the idea to be able to customize the call to `bookdown::render_book()`.

**Comment by cderv** [comment]
Maybe **pagedown** should make those formats book's format for **bookdown** ? There is a few thing that needs to be added I think so that it works as expected by **bookdown**. 

Using `render_book()` to use single format output but in multiple files is interesting. I don't think it is fully supported yet, and it works approximately now.
